### DISCUSSION TITLE
New store "on board" Infinispan cache

### DISCUSSION THREAD

**Discussion by LMonert** [keycloak/keycloak#15549] [Archive]
Hey everyone! I was reading the available documentation, discussions and blog posts regarding the new storage. However one thing is not quite clear to me. The legacy store comes with Infinispan cache "on board" which stores user sessions etc. and can be easily configured to share the cache across multiple instances. For the new store we will be able to mix storage types and explicitly decide what is stored in which storage type. We will also be able to use an external Infinispan and assign specific data to store in it. However there is no mention of the previously included "on board" Infinispan anymore. Will it be a requirement to setup an external Infinispan if we want to use cache with the new storage? Are there any concrete plans regarding this topic?

**Comment by hmlnarik** [comment]
Thank you for the question!

There are plans to support external infinispan storage only, no embedded infinispan is planned at this moment. This is to allow better experience for scaling so that Keycloak processing authentication requests and storage including cache can scale independently.

Support for a full-blown external cache service awaits tree storage implementation (#9742).
